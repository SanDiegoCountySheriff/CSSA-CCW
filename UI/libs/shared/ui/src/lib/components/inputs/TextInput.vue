<template>
  <v-text-field
    v-model="value"
    :label="$t(label)"
    :rules="rules"
    :name="name"
    @blur="handleBlur"
  />
</template>

<script setup lang="ts">
import { ref } from 'vue';

export interface TextInputProps {
  label?: string;
  name?: string;
  rules?: string[];
  target?: string;
}

const props = withDefaults(defineProps<TextInputProps>(), {
  label: '',
  name: '',
  rules: () => [],
  target: '',
});

const emit = defineEmits(['input']);

function handleBlur(event) {
  emit('input', event.target.value, props.target);
}

const value = ref('');
</script>
