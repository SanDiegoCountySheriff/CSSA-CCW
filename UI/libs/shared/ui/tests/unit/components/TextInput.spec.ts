import TextInput from '@shared-ui/components/inputs/TextInput.vue';
import Vuetify from 'vuetify';
import { createLocalVue, mount, shallowMount } from '@vue/test-utils';

const localVue = createLocalVue();
const tMock = {
  $t: text => text,
};

describe('TextInput', () => {
  let vuetify;

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  it('Should render the correct label', () => {
    //@ts-ignore
    const wrapper = shallowMount(TextInput, {
      localVue,
      vuetify,
      propsData: {
        label: 'testLabel',
      },
      mocks: tMock,
    });
    expect(wrapper.find('v-text-field-stub').attributes().label).toEqual(
      'testLabel'
    );
  });
  it('Should match snapshot', () => {
    //@ts-ignore
    const wrapper = mount(TextInput, {
      localVue,
      vuetify,
      mocks: tMock,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
