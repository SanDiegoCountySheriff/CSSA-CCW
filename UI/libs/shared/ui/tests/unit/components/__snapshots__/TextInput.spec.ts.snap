// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Should match snapshot 1`] = `
<div class="v-input theme--light v-text-field">
  <div class="v-input__control">
    <div class="v-input__slot">
      <div class="v-text-field__slot"><input name="" id="input-5" type="text"></div>
    </div>
    <div class="v-text-field__details">
      <div class="v-messages theme--light">
        <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
      </div>
    </div>
  </div>
</div>
`;
